src/escrows/LPCurveYearnV2Escrow.sol

```diff
@@ -73,31 +91,51 @@ contract DolaFraxBPEscrow {
     function pay(address recipient, uint amount) public {
         if (msg.sender != market) revert OnlyMarket();
         uint256 tokenBal = token.balanceOf(address(this));
-        uint256 missingAmount = amount > tokenBal ? amount - tokenBal : 0;
-        // If there are enough tokens in the escrow, transfer the amount
+
         if (tokenBal >= amount) {
             token.safeTransfer(recipient, amount);
-        } else if (stakedBalance >= missingAmount) {
-            // If there are enough staked tokens in convex, withdraw the amount from convex
-            stakedBalance -= missingAmount;
-            rewardPool.withdrawAndUnwrap(missingAmount, false);
-            token.safeTransfer(recipient, amount);
-        } else if (
-            YearnVaultV2Helper.collateralToAsset(
-                yearn,
-                yearn.balanceOf(address(this))
-            ) >= missingAmount
-        ) {
-            // If there are enough tokens in Yearn, withdraw the amount from Yearn
-            uint256 withdrawAmount = YearnVaultV2Helper.assetToCollateral(
+            return;
+        }
+
+        uint256 missingAmount = amount > tokenBal ? amount - tokenBal : 0;
+        uint256 convexBalance = IERC20(address(rewardPool)).balanceOf(
+            address(this)
+        );
+        if (convexBalance > 0 && missingAmount > 0) {
+            uint256 withdrawAmount = convexBalance > missingAmount
+                ? missingAmount
+                : convexBalance;
+            missingAmount -= withdrawAmount;
+            rewardPool.withdrawAndUnwrap(withdrawAmount, false);
+        }
+
+        uint yearnBal = yearn.balanceOf(address(this));
+        if (yearnBal > 0 && missingAmount > 0) {
+            uint256 maxWithdraw = YearnVaultV2Helper.collateralToAsset(
                 yearn,
-                missingAmount
+                yearnBal
             );
+            uint256 withdrawAmount = maxWithdraw > missingAmount
+                ? missingAmount
+                : maxWithdraw;
+
+            uint256 collateralAmount;
+            if (withdrawAmount == maxWithdraw) collateralAmount = yearnBal;
+            else
+                collateralAmount = YearnVaultV2Helper.assetToCollateral(
+                    yearn,
+                    withdrawAmount + weiDelta
+                );
             // Withdraw from Yearn
-            uint256 assetAmount = yearn.withdraw(withdrawAmount, address(this));
-            // Transfer the amount to the recipient
-            token.safeTransfer(recipient, assetAmount + tokenBal);
+            yearn.withdraw(collateralAmount, address(this));
+
+            uint256 lpToPay = token.balanceOf(address(this));
+            if (lpToPay != amount) {
+                _ensureLimitsOrRevert(lpToPay, amount);
+                amount = lpToPay;
+            }
```

Comment:
1. Is this check necessary? Looks to me as it'll revert on transfer if `lpToPay < amount`, and that there's no real issue if `lpToPay > weiDelta`.
2. There's no reason to pay the `weiDelta` as long as we have enough lp tokens to transfer `amount`.